src/app/screens/speedUpTransaction/index.tsx

```diff
@@ -0,0 +1,541 @@
+            : selectedAccount.id,
+        network: network.type,
+        seedVault,
+      });
+      setRbfTransaction(rbfTx);
+
+      const rbfTransactionSummary = await rbf.getRbfTransactionSummary(transaction);
+      setRbfTxSummary(rbfTransactionSummary);
+
+      const mempoolFees = await btcClient.getRecommendedFees();
+      setRecommendedFees(mempoolFees);
+
+      const rbfRecommendedFeesResponse = await rbfTx.getRbfRecommendedFees(mempoolFees);
+      setRbfRecommendedFees(rbfRecommendedFeesResponse);
+    } catch (err: any) {
+      console.error(err);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchRbfData();
+  }, [selectedAccount, id, transaction]);
+
+  const handleClickFeeButton = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (e.currentTarget.value === 'custom') {
+      setShowCustomFee(true);
+      return;
+    }
+
+    if (rbfRecommendedFees) {
+      const feeObj = rbfRecommendedFees[e.currentTarget.value];
+
+      if (feeObj?.enoughFunds) {
+        setTotalFee(feeObj.fee);
+        setFeeRateInput(feeObj.feeRate);
+        setSelectedOption(e.currentTarget.value);
+      }
+    }
+  };
+
+  const handleGoBack = () => {
+    navigate(-1);
+  };
+
+  const calculateTotalFee = async (feeRate: string) => {
+    const feeSummary: {
+      enoughFunds: boolean;
+      fee?: number;
+      feeRate: number;
+    } = await rbfTransaction.getRbfFeeSummary(Number(feeRate));
+
+    if (!feeSummary.enoughFunds) {
+      setCustomFeeError(t('INSUFFICIENT_FUNDS'));
+    } else {
+      setCustomFeeError(undefined);
+    }
+
+    if (feeSummary.fee) {
+      setCustomTotalFee(feeSummary.fee.toString());
+    }
+
+    return feeSummary.fee;
+  };
+
+  const signAndBroadcastTx = async (transport?: TransportType) => {
+    if (isLedgerAccount(selectedAccount) && !transport) {
+      return;
+    }
+
+    try {
+      const signedTx = await rbfTransaction.getReplacementTransaction({
+        feeRate: Number(feeRateInput),
+        ledgerTransport: transport,
+      });
+
+      await btcClient.sendRawTransaction(signedTx.hex);
+
+      toast.success(t('TX_FEE_UPDATED'));
+      handleGoBack();
+    } catch (err: any) {
+      console.error(err);
+
+      if (err?.response?.data && err?.response?.data.includes('insufficient fee')) {
+        toast.error('Insufficient fee');
+      }
+    }
+  };
+
+  const handleClickSubmit = async () => {
+    if (!selectedAccount || !id) {
+      return;
+    }
+
+    if (isLedgerAccount(selectedAccount)) {
+      setIsModalVisible(true);
+      return;
+    }
+
+    signAndBroadcastTx();
+  };
+
+  const handleConnectAndConfirm = async () => {
+    if (!selectedAccount) {
+      console.error('No account selected');
+      return;
+    }
+
+    setIsButtonDisabled(true);
+    const transport = await Transport.create();
+    if (!transport) {
+      setIsConnectSuccess(false);
+      setIsConnectFailed(true);
+      setIsButtonDisabled(false);
+    }
```

Comment:
Should there maybe be a return at the end of the if?

Reply:
Good catch, I think it should :)